File: api/seed.py
from db.seed import run_seed
from api.responses import safe


def seed_sample_data():
    return safe(lambda: _seed_sample_data())


def _seed_sample_data():
    run_seed(sample_data=True)

    return {
        "status": "seeded",
        "message": "Demo sample data seeded successfully.",
        "summary": {
            "accounts": 4,
            "budgets": 3,
            "recurring_rules": 4,
            "goals": 3,
            "transactions": 30,
        },
    }

File: frontend/src/lib/api/seed.ts
import { callApi } from './client';

export type SeedSampleDataResult = {
	status: string;
	message: string;
	summary: {
		accounts: number;
		budgets: number;
		recurring_rules: number;
		goals: number;
		transactions: number;
	};
};

export function seedSampleData() {
	return callApi<SeedSampleDataResult>('seed_sample_data');
}